### PR TITLE
refactor: update for go mod, fmt, publish Darwin distro, and update sample config for success

### PR DESCRIPTION
**Pull request by jaredscheib**
This PR:
1. refactors the codebase and README to use go 1.14 + go mod, and to publish a macOS distro
1. applies go.fmt on the codebase
1. adds additional request/response debugging
1. adds http request headers: a) content type so that requests succeed; b) identify the user agent
1. adds `team_id` to the sample `main.tf` script to ensure that requests succeed (avoiding 502)
1. indicates which of the fields for the machine create path have been verified to work by me so far (there's not reason to believe the others don't, but my purpose in this work was not to test everything, and i didn't test those yet)
1. updates the sample `main.tf` to use the latest ubuntu template version we offer
1. accommodates renaming the repo to `terraform-provider-paperspace` to align with convention for terraform providers

I published the macOS distro and rebuilt & published the updated linux distro as well. I left the Windows distro as-is.

This was done in support of my work on https://paperspace.atlassian.net/browse/PS-13785.

I didn't update the linux or Windows distros or test the rest of the code because that's not being asked for right now

### CODE REVIEW

File: src/terraform-provider-paperspace/main.tf [hunk 1]
```diff
@@ -31,7 +32,8 @@ resource "paperspace_machine" "my-machine-1" {
   size = 50
   billing_type = "hourly"
   assign_public_ip = true // optional, remove if you don't want a public ip assigned
-  template_id = "${data.paperspace_template.my-template-1.id}"
-  user_id = "${data.paperspace_user.my-user-1.id}"  // optional, remove to default
-  script_id = "${paperspace_script.my-script-1.id}" // optional, remove for no script
+  template_id = data.paperspace_template.my-template-1.id
+  user_id = data.paperspace_user.my-user-1.id  // optional, remove to default
+  team_id = data.paperspace_user.my-user-1.team_id
```
Comment (jaredscheib) [inline src/terraform-provider-paperspace/main.tf:37]: with this missing, the api would respond with a 502

File: src/terraform-provider-paperspace/main.tf [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-  email = "me@mycompany.com" // change to the email address of a user on your paperspace team
+data "paperspace_template" "my-template-1" {
+  label = "Ubuntu 18.04 Server"
 }
 
-data "paperspace_template" "my-template-1" {
-  label = "Ubuntu 16.04 Server"
+data "paperspace_user" "my-user-1" {
+  email = "user@email.com"
+  team_id = "te1234567"
```
Comment (jaredscheib) [inline src/terraform-provider-paperspace/main.tf:12]: with `team_id` missing, the api would respond with a 502

File: src/terraform-provider-paperspace/main.tf [hunk 3]
```diff
@@ -1,14 +1,15 @@
 provider "paperspace" {
-  api_key = "1be4f97..." // modify this to use your actual api key
   region = "East Coast (NY2)"
+  api_key = "12345678..."
 }
 
-data "paperspace_user" "my-user-1" {
-  email = "me@mycompany.com" // change to the email address of a user on your paperspace team
+data "paperspace_template" "my-template-1" {
+  label = "Ubuntu 18.04 Server"
```
Comment (jaredscheib) [inline src/terraform-provider-paperspace/main.tf:7]: update to latest ubuntu template we currently offer

File: src/terraform-provider-paperspace/main.tf [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 }
 
-data "paperspace_template" "my-template-1" {
-  label = "Ubuntu 16.04 Server"
+data "paperspace_user" "my-user-1" {
+  email = "user@email.com"
```
Comment (jaredscheib) [inline src/terraform-provider-paperspace/main.tf:11]:
```suggestion
  email = "me@mycompany.com" // change to the email address of a user on your paperspace team
```

File: src/terraform-provider-paperspace/main.tf [hunk 5]
```diff
@@ -1,14 +1,15 @@
 provider "paperspace" {
-  api_key = "1be4f97..." // modify this to use your actual api key
   region = "East Coast (NY2)"
+  api_key = "12345678..."
```
Comment (jaredscheib) [inline src/terraform-provider-paperspace/main.tf:3]:
```suggestion
  api_key = "1be4f97..." // modify this to use your actual api key
```

File: src/terraform-provider-paperspace/client.go [hunk 6]
```diff
@@ -56,37 +57,86 @@ type Config struct {
...
+		return nil
+	})
+
+	restyClient.
+		SetHostURL(c.ApiHost).
+		SetHeader("x-api-key", c.ApiKey).
+		SetHeader("Accept", "application/json").
+		SetHeader("Content-Type", "application/json").
+		SetHeader("User-Agent", "terraform-provider-paperspace").
+		SetHeader("ps_client_name", "terraform-provider-paperspace")
```
Comment (weikaolun) [inline src/terraform-provider-paperspace/client.go:122]: 😍

**Pull request merged**